File: 349-intersection-of-two-arrays/intersection-of-two-arrays.ts
function intersection(nums1: number[], nums2: number[]): number[] {
    let arr:number[]=[]
    for (let i = 0; i < nums2.length; i++) {
            if (nums1.includes(nums2[i])&& !arr.includes(nums2[i])) {
                arr.push(nums2[i])
            }
        
    }
    // for (let j = 0; j < arr.length; j++) {
    //     for (let k = j+1; k < arr.length; k++) {
    //         if ( arr[j] === arr[k]) { 
    //             arr.splice(k,1, )
    //     }
    // }
    // }
    return arr
};

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()
        intersection = []
        i, j = 0, 0
        while i < len(nums1) and j < len(nums2):
            if nums1[i] == nums2[j] and nums1[i] not in intersection:
                intersection.append(nums1[i])
                i += 1
                j += 1
            elif nums1[i] < nums2[j]:
                i += 1
            else:
                j += 1
        return intersection